msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/ltw/TestCase3029738.java

```diff
@@ -0,0 +1,145 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.List;
+
+// Sign in with AAD and MSA account
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/3029738
+@LTWTests
+@RetryOnFailure
+@SupportedBrokers(brokers = {BrokerLTW.class})
+@RunWith(Parameterized.class)
+public class TestCase3029738 extends AbstractMsalBrokerTest {
+
+    private final UserType mUserType;
+
+    public TestCase3029738(@NonNull UserType userType) {
+        mUserType = userType;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static List<UserType> userType() {
+        return Arrays.asList(
+                UserType.MSA,
+                UserType.CLOUD
+        );
+    }
+
+    @Test
+    public void test() throws Throwable {
+        // Check flight, this is checking what was passed to automation app, not the broker apks
+        Assume.assumeTrue( "EnableSystemAccountManager flight is not activated, Test will be skipped",
+                BuildConfig.COPY_OF_LOCAL_FLIGHTS_FOR_TEST_PURPOSES.contains("EnableSystemAccountManager:true"));
+
+        // Fetch account credentials
+        final String username = mLabAccount.getUsername();
+        final String password = mLabAccount.getPassword();
+
+        // Install and launch msal test app
+        // set configuration based on user type
+        final MsalTestApp msalTestApp = new MsalTestApp();
+        msalTestApp.install();
+        msalTestApp.launch();
+        msalTestApp.handleFirstRunBasedOnUserType(mUserType);
+
+        // Prompt handler for the subsequent
+        final MicrosoftStsPromptHandlerParameters promptHandlerParameters = MicrosoftStsPromptHandlerParameters.builder()
+                .prompt(PromptParameter.SELECT_ACCOUNT)
+                .loginHint(username)
+                .sessionExpected(false)
+                .broker(mBroker)
+                .expectingBrokerAccountChooserActivity(false)
+                .expectingProvidedAccountInBroker(false)
+                .expectingLoginPageAccountPicker(false)
+                .expectingProvidedAccountInCookie(false)
+                .consentPageExpected(false)
+                .passwordPageExpected(true)
+                .speedBumpExpected(false)
+                .registerPageExpected(false)
+                .enrollPageExpected(false)
+                .staySignedInPageExpected(false)
+                .verifyYourIdentityPageExpected(false)
+                .howWouldYouLikeToSignInExpected(false)
+                .build();
+
+        // Make sure we get a token
+        final String token = msalTestApp.acquireToken(username, password, promptHandlerParameters, true);
+        Assert.assertNotNull(token);
+
+        // Launch OS Account page and make sure username shows up in account manager.
+        getSettingsScreen().launchAccountListPage();
+        Assert.assertTrue(UiAutomatorUtils.obtainUiObjectWithText(username).exists());
```

Comment:
Should we add another check that they show up under correct Account Type.

Reply:
Sure thing, ill take a look

Reply:
Added in https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/2207/commits/deefc664d8a6cd8a7a099c9da7d1ee4bbad16695